src/towncrier/test/test_hg.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Amber Brown, 2015
+# See LICENSE for details.
+
+
+from pathlib import Path
+from subprocess import check_call
+
+from click.testing import CliRunner
+from twisted.trial.unittest import TestCase
+
+from towncrier import _hg, _vcs
+
+from .helpers import setup_simple_project, write
+
+
+def create_project(
+    pyproject_path="pyproject.toml", main_branch="main", extra_config=""
+):
+    setup_simple_project(pyproject_path=pyproject_path, extra_config=extra_config)
+    Path("foo/newsfragments/123.feature").write_text("Adds levitation")
+    initial_commit(branch=main_branch)
+    check_call(["hg", "branch", "otherbranch"])
+
+
+def initial_commit(branch="main"):
+    """
+    Create a mercurial repo, configure it and make an initial commit
+
+    There must be uncommitted changes
+    """
+    check_call(["hg", "init", "."])
+    check_call(["hg", "branch", branch])
+    commit("Initial Commit")
+
+
+def commit(message):
+    """Commit the repo in the current working directory
+
+    There must be uncommitted changes
+    """
+    check_call(["hg", "addremove", "."])
+    check_call(["hg", "commit", "-m", message])
+
+
+class TestHg(TestCase):
+    def test_get_default_compare_branch(self):
+        assert _hg.get_default_compare_branch(["main", "default"]) == "default"
+        assert _hg.get_default_compare_branch(["main", "a_topic"]) is None
+
+    def test_empty_remove(self):
+        """
+        If remove_files gets an empty list, it returns gracefully.
+        """
+        _hg.remove_files([])
+
+    def test_hg(self):
```

Comment:
can we have a docstring for these tests describing what is the scope/goal of each test?

What is the different between `test_hg` and `test_get_default_compare_branch` ?

We don't have any other tools to validated the code inside the tests.
So a good description can help a reviewer validated whether the code inside the test function is valid or not